src/react/WithQueries.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { Function1 } from 'fp-ts/lib/function';
+import { declareQueries } from './declareQueries';
+import { QueryResult } from '../QueryResult';
+import {
+  EnforceNonEmptyRecord,
+  ObservableQueries,
+  ProductL,
+  ProductP,
+  ProductA
+} from '../util';
+
+type Params<A> = A extends void
+  ? {}
+  : {
+      params: A;
+    };
+
+type Props<R extends ObservableQueries> = {
+  queries: EnforceNonEmptyRecord<R>;
+  render: Function1<QueryResult<ProductL<R>, ProductP<R>>, React.ReactNode>;
+} & Params<ProductA<R>>;
+
+export function WithQueries<P extends ObservableQueries>(props: Props<P>) {
+  const WrappedComponent = React.useMemo(
+    () =>
+      declareQueries(props.queries)(({ queries }) => (
+        <>{props.render(queries)}</>
+      )),
+    Object.values(props.queries)
```

Comment:
I added it but I have no idea on how to test this.. I wonder if we should actually, wouldn't we be testing our own understanding of `useMemo`? (which by the way seems fair as I never used it before, but maybe doesn't deserve a unit test?)

Reply:
I'd say it's ok if we "test this in prod" for the moment, aka if you rebase this branch you should be able to test it on a real project. Let me know how it goes ;)
Other than this, I think the PR is good to go 👍

Reply:
I tested and dynamic selection of queries breaks the type checking in an unpredicted way...
should be further investigated if we want to add this possibility

Reply:
interesting, can you share snippet?

Reply:
sure:
```
  changeState = () => this.setState(false);

  renderImage() {
    return (
      <>
        <Button onClick={this.changeState} />
        <WithQueries
          queries={this.state ? { presignedLink } : { presignedLink, userDocs }}
          params={{
            presignedLink: {
              docName: this.props.link,
              userDocs: { filter: undefined, limit: undefined }
            }
          }}
          render={queries =>
            queries.fold(
              () => null,
              () => null,
              a => {
                return (
                  <>
                    {JSON.stringify(a)}
                    <img src={a.presignedLink} width="400px" height="500px" />
                  </>
                );
              }
            )
          }
        />
      </>
    );
  }
```
where presignedLink is a `Query` with `_A = { docName: string } and `userDocs` is a `Query` with `{ filter?: string | undefined; limit?: number | undefined;}`.

I wonder if dynamic queries work with `declareQueries` altogether?

Reply:
> breaks the type checking

and what is the type error?

Reply:
I confirm that also `declareQueries` type checks fail strangely when changing dynamically the queries (fallbacks to `never`)

Reply:
> and what is the type error?

`params` is set to `never` although `presignedLink` has `_A = { docName: string }`

Reply:
Ok then, let's ignore this "feature" and thus remove the `Object.values`. One can always conditionally render two different `<WithQueries ... />` based on a condition

Reply:
that was exactly my thought 👍

Reply:
maybe we could explicitly write in the readme that it is not supported?

Reply:
yes 👍